.github/workflows/openai.yml

```diff
@@ -5,20 +5,20 @@ name: OpenAI
 
 on:
   pull_request_target:
-    branches: ['main']
+    branches: ["main"]
     paths:
-      - 'autogen/**'
-      - 'test/**'
-      - 'notebook/agentchat_auto_feedback_from_code_execution.ipynb'
-      - 'notebook/agentchat_function_call.ipynb'
-      - '.github/workflows/openai.yml'
+      - "autogen/**"
+      - "test/**"
+      - "notebook/agentchat_auto_feedback_from_code_execution.ipynb"
+      - "notebook/agentchat_function_call.ipynb"
+      - ".github/workflows/openai.yml"
 
 jobs:
   test:
     strategy:
       matrix:
         os: [ubuntu-latest]
-        python-version: ["3.9", "3.10", "3.11"]
+        python-version: ["3.9", "3.10", "3.11", "3.12"]
```

Comment:
Please check test_notebook.py and check the condition to run each notebook. Redistribute the load after 3.12 is added.